applications/nrf9160dk_demo/src/main.c

```diff
@@ -0,0 +1,1029 @@
+/**brief Initialize LTE status containers. */
+static void modem_data_init(void)
+{
+	int err;
+
+	err = modem_info_init();
+	if (err) {
+		LOG_ERR("Modem info could not be established: %d", err);
+		return;
+	}
+
+	modem_info_params_init(&modem_param);
+
+	signal_strength_cloud_data.type = CLOUD_CHANNEL_LTE_LINK_RSRP;
+	signal_strength_cloud_data.tag = 0x1;
+
+	modem_info_rsrp_register(modem_rsrp_handler);
+}
+#endif /* CONFIG_MODEM_INFO */
+
+/**@brief Initializes the sensors that are used by the application. */
+static void sensors_init(void)
+{
+#if CONFIG_MODEM_INFO
+	modem_data_init();
+#endif /* CONFIG_MODEM_INFO */
+
+	k_work_submit_to_queue(&application_work_q, &device_status_work);
+
+	button_sensor_init();
+}
+
+/**@brief User interface event handler. */
+static void ui_evt_handler(struct ui_evt evt)
+{
+	button_send(evt.button, evt.type == UI_EVT_BUTTON_ACTIVE ? 1 : 0);
+}
+
+void handle_bsdlib_init_ret(void)
+{
+	#if defined(CONFIG_BSD_LIBRARY)
+	int ret = bsdlib_get_init_ret();
+
+	/* Handle return values relating to modem firmware update */
+	switch (ret) {
+	case MODEM_DFU_RESULT_OK:
+		LOG_INF("MODEM UPDATE OK. Will run new firmware");
+		sys_reboot(SYS_REBOOT_COLD);
+		break;
+	case MODEM_DFU_RESULT_UUID_ERROR:
+	case MODEM_DFU_RESULT_AUTH_ERROR:
+		LOG_ERR("MODEM UPDATE ERROR %d. Will run old firmware", ret);
+		sys_reboot(SYS_REBOOT_COLD);
+		break;
+	case MODEM_DFU_RESULT_HARDWARE_ERROR:
+	case MODEM_DFU_RESULT_INTERNAL_ERROR:
+		LOG_ERR("MODEM UPDATE FATAL ERROR %d. Modem failiure", ret);
+		sys_reboot(SYS_REBOOT_COLD);
+		break;
+	default:
+		break;
+	}
+	#endif /* CONFIG_BSD_LIBRARY */
+}
+
+void main(void)
+{
+	int ret;
+
+	LOG_INF("nRF9160DK Demo Started");
+	k_work_q_start(&application_work_q, application_stack_area,
+		       K_THREAD_STACK_SIZEOF(application_stack_area),
+		       CONFIG_APPLICATION_WORKQUEUE_PRIORITY);
+	if (IS_ENABLED(CONFIG_WATCHDOG)) {
+		watchdog_init_and_start(&application_work_q);
+	}
+	handle_bsdlib_init_ret();
+
+	cloud_backend = cloud_get_binding("NRF_CLOUD");
+	__ASSERT(cloud_backend != NULL, "nRF Cloud backend not found");
+
+	ret = cloud_init(cloud_backend, cloud_event_handler);
+	if (ret) {
+		LOG_ERR("Cloud backend could not be initialized, error: %d",
+			ret);
+		cloud_error_handler(ret);
+	}
+
+	ui_init(ui_evt_handler);
+
+	ret = cloud_decode_init(cloud_cmd_handler);
+	if (ret) {
+		LOG_ERR("Cloud command decoder initialization error: %d", ret);
+		cloud_error_handler(ret);
+	}
+
+	work_init();
+	modem_configure();
+connect:
+	ret = cloud_connect(cloud_backend);
+	if (ret != CLOUD_CONNECT_RES_SUCCESS) {
+		cloud_connect_error_handler(ret);
+	} else {
+		atomic_set(&reconnect_to_cloud, 0);
+		k_delayed_work_submit_to_queue(&application_work_q,
+					       &cloud_reboot_work,
+					       CLOUD_CONNACK_WAIT_DURATION);
+	}
+
+	struct pollfd fds[] = {
+		{
+			.fd = cloud_backend->config->socket,
+			.events = POLLIN
+		}
+	};
+
+	while (true) {
+		ret = poll(fds, ARRAY_SIZE(fds),
```

Comment:
should this use https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2177 or leave polling in main?

Reply:
#2177 looks helpful, but should I wait for event manager integration?

Reply:
i do not know the answer

Reply:
I would not wait on event manger. It will take quite some time before we have anything useful based on that.

Reply:
@jayteemo are you suggesting that I wait for #2177 to be approved and merged, or incorporate it before then?  seems better to wait (needs just 1 more approval)

Reply:
i wasn't suggesting either, just asking if you thought it should be used in this new application.  i agree, probably best to wait if you wish to use it.